mg/builtins.go

```diff
@@ -31,14 +32,19 @@ func (bcl BultinCmdList) Lookup(name string) (cmd BultinCmd, found bool) {
 	panic("internal error: the `.exec` BuiltinCmd is not defined")
 }
 
+// BuiltinCmds is used to add a list of predefined commands to the state. It
+// implements Reducer interface.
 type BuiltinCmds struct{}
 
+// ExecCmd spawns the given command in a goroutine and returns its State right
```

Comment:
I'd just say something like `ExecCmd implements the \`.exec\` builtin`. It's ran in a reducer and the command can take a while to run so it's already implied that it's non-blocking unless documented otherwise.